test-app-reactive/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/integration/HmppsEventProcessingTest.kt

```diff
@@ -115,4 +124,52 @@ class HmppsEventProcessingTest : IntegrationTestBase() {
     assertThat(receivedEvent.type).isEqualTo("FIFO-EVENT")
     assertThat(receivedEvent.contents).isEqualTo("some FIFO contents")
   }
+
+  @Test
+  fun `large message pointer is published to topic and message body stored in S3`() = runTest {
+    val file = File("src/test/resources/events/large-sns-message.json").bufferedReader().readLines()
+    val event = HmppsEvent("fifo-event-id", "FIFO-EVENT", file.toString())
+    largeMessageFifoTopic.publish(
+      eventType = event.type,
+      event = gsonString(event),
+      messageGroupId = UUID.randomUUID().toString(),
+    )
+
+    await untilCallTo { largeMessageFifoSqsClient.countMessagesOnQueue(largeMessageFifoQueueUrl).get() } matches { it == 1 }
+
+    val (message) = ReceiveMessageRequest.builder().queueUrl(largeMessageFifoQueueUrl).build()
+      .let { largeMessageFifoSqsClient.receiveMessage(it).get().messages()[0].body() }
+      .let { objectMapper.readValue(it, Message::class.java) }
+
+    val s3Reference: java.util.ArrayList<*> = objectMapper.readValue(message, ArrayList::class.java)
+    val s3Pointer = objectMapper.readValue(objectMapper.writeValueAsString(s3Reference[1]), LinkedHashMap::class.java)
+    assertThat(s3Pointer.get("s3BucketName")).isEqualTo("bucket-name")
+    assertThat(s3Pointer.keys).contains("s3Key")
+    val s3Object = s3Client.getObject(
+      GetObjectRequest.builder().bucket("bucket-name").key(s3Pointer.get("s3Key").toString()).build(),
+      AsyncResponseTransformer.toBytes(),
+    ).join()
```

Comment:
Would clients of the library have to do this after reading a message off the queue? Just wondering if we could / should hide this complexity in the library - maybe hanging off HmppsQueue or something?

Reply:
If we do the SQS large message in this PR it will be pretty massive - do you want that or should we go with this first and do the SQS next?

Reply:
Decided to do SQS client in a separate PR